docs/adr/0005-api-for-spaces.md

```diff
@@ -0,0 +1,107 @@
+# 5. Graph API for Spaces
+
+* Status: proposed
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1
+* Date: 2021-03-19
+
+Technical Story: API to enable the concept of [Spaces](https://github.com/owncloud/enterprise/issues/3863)
+
+## Context and Problem Statement
+
+As one of the building blocks for Spaces in oCIS we plan to add an API that returns information about available spaces.
+
+> Note: The term "spaces" is used here in the context of "a space where files can be saved", similar to a directory. It is not to be confused with space in the sense of file space for example.
+
+The purpose of this new API is to give clients a very simple way to query the dynamic list of spaces, that the user has access to. Clients can provide a way better user experience with that.
+
+This API would even allow to provide (WebDAV-) endpoints depending on the kind and version of the client asking for it.
```

Comment:
> the kind and version of the client asking for it.

What does it mean?

Reply:
IMO that means the "new" clients would ask for the info, the legacy ones do not know about the api.

Reply:
If we wanted (or have to) we could consider in the backend code, _which_ client in _which_ version is asking and tweak the answer accordingly. I am not saying that we want to do that per default. But it could be an SOS in some situations.